targets/TARGET_STM/TARGET_STM32L4/STM32Cube_FW/CMakeLists.txt

```diff
@@ -1,15 +1,120 @@
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_pwr_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_qspi.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_rcc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_rcc_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_rng.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_rng_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_rtc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_rtc_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_sai.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_sai_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_sd.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_sd_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_smartcard.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_smartcard_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_smbus.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_spi.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_spi_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_sram.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_swpmi.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_tim.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_tim_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_tsc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_uart.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_uart_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_usart.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_usart_ex.c
+        STM32L4xx_HAL_Driver/stm32l4xx_hal_wwdg.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_adc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_comp.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_crc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_crs.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_dac.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_dma2d.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_dma.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_exti.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_fmc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_gpio.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_i2c.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_lptim.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_lpuart.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_opamp.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_pka.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_pwr.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_rcc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_rng.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_rtc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_sdmmc.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_spi.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_swpmi.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_tim.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_usart.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_usb.c
+        STM32L4xx_HAL_Driver/stm32l4xx_ll_utils.c
+
+        STM32L4xx_HAL_Driver/Legacy/stm32l4xx_hal_can_legacy.c
 )
 
 target_include_directories(mbed-core
     INTERFACE
-        ${CMAKE_CURRENT_SOURCE_DIR}
-	CMSIS
+        .
+        CMSIS
+        STM32L4xx_HAL_Driver
+        STM32L4xx_HAL_Driver/Legacy
```

Comment:
F4 uses `${CMAKE_CURRENT_SOURCE_DIR}/` -- I think we should be consistent.

I personally prefer without the `${CMAKE_CURRENT_SOURCE_DIR}`

Reply:
We will refactor entire codebase (I only changed what I touched here)

Reply:
👍